components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AzureOpenAIEmbeddingProviderServiceImpl.java

```diff
@@ -0,0 +1,110 @@
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package org.wso2.carbon.apimgt.impl;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.util.EntityUtils;
+import org.osgi.service.component.annotations.Component;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.api.EmbeddingProviderService;
+import org.wso2.carbon.apimgt.impl.utils.APIUtil;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+
+@Component(
+        name = "azure.openai.embedding.provider.service",
+        immediate = true,
+        service = EmbeddingProviderService.class
+)
+public class AzureOpenAIEmbeddingProviderServiceImpl implements EmbeddingProviderService {
+
+    private HttpClient httpClient;
+    private String azureApiKey;
+    // e.g., https://<your-resource-name>.openai.azure.com/openai/deployments/<deployment-id>/embeddings?api-version=2024-02-15-preview
+    private String endpointUrl;
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    @Override
+    public void init(Map<String, String> providerConfig) throws APIManagementException {
+        this.azureApiKey = providerConfig.get("apikey");
+        this.endpointUrl = providerConfig.get("embedding_endpoint");
+
+        if (this.azureApiKey == null || this.endpointUrl == null) {
+            throw new APIManagementException(
+                    "Missing required Azure OpenAI configuration properties: 'apikey' and/or 'embedding_endpoint'");
+        }
+
+        this.httpClient = APIUtil.getHttpClient(endpointUrl);
+    }
+
+    @Override
+    public String getType() {
+        return "AZURE_OPENAI";
+    }
+
+    @Override
+    public int getEmbeddingDimension() throws APIManagementException {
+        return getEmbedding(this.getType()).length;
+    }
+
+    @Override
+    public float[] getEmbedding(String input) throws APIManagementException {
+        HttpPost post = new HttpPost(endpointUrl);
+        post.setHeader("api-key", azureApiKey);
+        post.setHeader("Content-Type", "application/json");
```

Comment:
Shall we use constants for "Content-Type", "application/json" and "api-key" strings?